tiledb/sm/filter/filter_pipeline.h

```diff
@@ -92,9 +101,10 @@ class FilterPipeline {
    * Populates the filter pipeline from the data in the input binary buffer.
    *
    * @param buff The buffer to deserialize from.
-   * @return Status
+   * @return Status and FilterPipeline
    */
-  Status deserialize(ConstBuffer* buff);
+  static std::tuple<Status, optional<std::shared_ptr<FilterPipeline>>>
+  deserialize(ConstBuffer* buff);
```

Comment:
I don't think the `shared_ptr` is extraneous here. Every place that's calling `deserialize` is immediately deferencing the pointer (with `operator*()`). The intermediate allocation doesn't seem to be doing anything.

Reply:
changed to return std::tuple<Status, optional<FilterPipeline>>